File: php-pattern/basic/adapter/engine/Printer.php
<?php

namespace app\adapter\engine;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * Description of Printer
 *
 * @author eko
 */
class Printer {

    public function doPrint(Printable $printable) {
        echo $printable->printing();
    }

}

File: java-pattern/src/main/java/adapter/engine/Printer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package adapter.engine;

/**
 *
 * @author eko
 */
public class Printer {

    public void doPrint(Printable printable) {
        if (null != printable) {
            System.out.println("Printing => " + printable.printing());
        }
    }
}
